### PR TITLE
update dependencies: - NN 125.0.0; - common 23.3.0-rc.1; - maps 10.11.0-rc.1; - mapbox-java 6.11.0-beta.1

### PR DESCRIPTION
**Pull request by RingerJK**
### Description
update dependencies: - NN 125.0.0; - common 23.3.0-rc.1; - maps 10.11.0-rc.1; - mapbox-java 6.11.0-beta.1

~⚠️ `NavigationViewLifecycleTest` was disabled, description see in `Ignore` annotation in the code~

### CODE REVIEW

File: libnavigation-core/src/androidTest/java/com/mapbox/navigation/core/trip/service/NativeNavigatorCallbackOrderTest.kt [hunk 1]
```diff
@@ -176,7 +176,7 @@ internal class NativeNavigatorCallbackOrderTest :
         callbackInvocations.waitUntilHas(CallbackInvocation.Refresh, elementsAfter = 1)
 
         callbackInvocations.checkThatHas(
-            CallbackInvocation.Status(NavigationStatusOrigin.SET_ROUTE)
+            CallbackInvocation.Status(NavigationStatusOrigin.ROUTE_REFRESH)
```
Comment (dzinad) [inline libnavigation-core/src/androidTest/java/com/mapbox/navigation/core/trip/service/NativeNavigatorCallbackOrderTest.kt:179]: Hooray!

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/CoreRerouteTest.kt [hunk 2]
```diff
@@ -352,13 +352,13 @@ class CoreRerouteTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.jav
         rerouteStateTransitionAssertion.assert()
     }
 
-    @Test(timeout = 10_000)
+    @Test(timeout = 100_000)
```
Comment (dzinad) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/CoreRerouteTest.kt:355]: That's not needed.

File: changelog/unreleased/bugfixes/6928.md [hunk 3]
```diff
@@ -0,0 +1 @@
+- Fixed upcoming `RouteObject` incorrect location.
```
Comment (LukasPaczos) [inline changelog/unreleased/bugfixes/6928.md:1]: This fix was alread cherry-picked to v2.10, wasn't it? In this case we can skip the changelog entry.

File: changelog/unreleased/other/6928.md [hunk 4]
```diff
@@ -0,0 +1 @@
+- :warning: Knows issue: the sdk might crash when no locations are available. The fix will be in the upcoming releases.
```
Comment (LukasPaczos) [inline changelog/unreleased/other/6928.md:1]: Is there a workaround we could do?

Comment (dzinad) [inline changelog/unreleased/other/6928.md:1]: Maybe we'll just use Core SDK version that doesn't crash?..

File: changelog/unreleased/bugfixes/6928.md (hunk 3 above)
Comment (RingerJK) [reply changelog/unreleased/bugfixes/6928.md:1]: seems so https://github.com/mapbox/mapbox-navigation-android/releases/tag/v2.10.1

File: changelog/unreleased/other/6928.md (hunk 4 above)
Comment (RingerJK) [reply changelog/unreleased/other/6928.md:1]: I'm waiting for the reply from Ramon about it. But not sure: the crash produces under the hood in the common sdk.

Comment (RingerJK) [reply changelog/unreleased/other/6928.md:1]: different versions might have breaking changes, so we must descend them between different SDKs. As far as I remember rc-alpha-beta version might have breaking changes in every iteration. @LukasPaczos please correct me if I'm wrong

Comment (LukasPaczos) [reply changelog/unreleased/other/6928.md:1]: Or force an early upgrade to v23.3.1 stable which is already out. Looking at the changelog the versions should be compatible. Could you try this @RingerJK?

Comment (dzinad) [reply changelog/unreleased/other/6928.md:1]: Seriously, there might be breaking changes between rc.1 and stable?

Comment (RingerJK) [reply changelog/unreleased/other/6928.md:1]: I can, but even if everything compiled, methods incompatibility might produce a crash in runtime

Comment (RingerJK) [reply changelog/unreleased/other/6928.md:1]: >Seriously, there might be breaking changes between rc.1 and stable?

event between bug-fix versions 😄

Comment (dzinad) [reply changelog/unreleased/other/6928.md:1]: Let's ask NN to bump?

Comment (RingerJK) [reply changelog/unreleased/other/6928.md:1]: it's Friday evening, not the best time for that 😞 

I make smoke runs with common `23.3.1`, it works for me. wdyt folks, should we release with that version? I'd say it's ok for `RC` @dzinad @LukasPaczos

Comment (LukasPaczos) [reply changelog/unreleased/other/6928.md]: :shipit:

**Review by Zayankovsky — APPROVED** [review]

**Pull request merged**